packages/backend/src/crawling/Crawler.ts

```diff
@@ -0,0 +1,23 @@
+import { resolve } from "path";
+
+import Piscina from "piscina";
+import type { EntityManager } from "typeorm";
+
+import { filename } from "./worker";
+import type { Execution } from "../entity/Execution";
+
+export class Crawler {
+  static workerPool = new Piscina({
+    filename: resolve(__dirname, "./workerWrapper.js"),
+    workerData: { fullpath: filename },
+  });
+
+  static async crawl(execution: Execution) {
+    await this.workerPool.run(
+      {
+        executionId: execution.id,
+      },
+      { name: "run" },
+    );
+  }
```

Comment:
Creates a "worker pool" (an abstraction over the Node.JS worker threads) with the given file as the "multithreaded" code.

Calling `workerPool.run` with parameters (`executionId`) runs the given file in a separate thread (worker thread).